frontend/src/metabase/actions/components/ActionForm/ActionForm.tsx

```diff
@@ -154,37 +159,28 @@ export const ActionForm = ({
     );
   }
 
-  const hasFormFields = !!form.fields.length;
-
   return (
     <FormProvider
       initialValues={initialValues}
       validationSchema={formValidationSchema}
       onSubmit={handleSubmit}
       enableReinitialize
     >
-      {({ dirty }) => (
-        <Form
-          disabled={!dirty && hasFormFields}
-          role="form"
-          data-testid="action-form"
-        >
-          {form.fields.map(field => (
-            <FormFieldWidget key={field.name} formField={field} />
-          ))}
-
-          <ActionFormButtonContainer>
-            {onClose && <Button onClick={onClose}>{t`Cancel`}</Button>}
-            <FormSubmitButton
-              disabled={!dirty && hasFormFields}
-              title={submitTitle ?? t`Submit`}
-              {...submitButtonVariant}
-            />
-          </ActionFormButtonContainer>
-
-          <FormErrorMessage />
-        </Form>
-      )}
+      <Form role="form" data-testid="action-form">
```

Comment:
Removed `dirty` check

Reply:
I know it's not something this PR is doing, but doesn't `Form` have a `role="form"` out of the box?

Reply:
Also, should we better check for `isValid` for the `disabled` prop?

Reply:
the form itself and the submit button are disabled when the form is invalid. Or are you talking about something else?

Reply:
removed `role="form"` btw - there is a native form underneath, it's not needed

Reply:
> the form itself and the submit button are disabled when the form is invalid. Or are you talking about something else?

Hm, I think I've seen we used `isValid` somewhere for the `disabled` prop, maybe I got confused 🤔